### DISCUSSION TITLE
Logging in Google Cloud Logging

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#8643] [Archive]
Hi all,

Instead of setting up my own logger in custom extensions, I want to use the default pino logger from the context. However, it would be great if we can map pino levels, so they show up correctly in Google Cloud Platform's cloud logging.

This is described here: https://github.com/pinojs/pino/blob/master/docs/help.md#stackdriver

However, I'm unsure on where/how to make these changes in Directus. Is there a way at the moment to do this?

Cheers!

Kees.

**Comment by rijkvanzanten** [comment]
> However, it would be great if we can map pino levels, so they show up correctly in Google Cloud Platform's cloud logging.

In order to do this, you'd have to override the actual Pino options that are defined here:

https://github.com/directus/directus/blob/fc87c0b662903214ca301e0a6b4af21efe135422/api/src/logger.ts#L7-L18

That's where you'd add in those formatters for "level" and "log" to allow google cloud platform to pick it up.

I think this could be a great point for another "init" hook (cc @nickrum). That would allow you to add a "init pino" hook that would then allow you to inject anything in the pino configuration when it's being created 🙂

**Reply by keesvanbemmel** [reply]
Thanks Rijk, as always.

I don't see a straightforward way to override that class atm in our setup where we just list Directus as a regular dependency. I'll look out for other options as well as your suggested hook!

**Comment by keesvanbemmel** [comment]
@nickrum I would love to create a PR for this but struggling with how to start. Any pointers?
